compiler/src/yul/operations.rs

```diff
@@ -111,3 +94,38 @@ pub fn decode(typ: FixedSize, ptr: yul::Expression) -> yul::Expression {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::yul::operations::{
+        emit_event,
+        encode,
+    };
+    use fe_semantics::namespace::events::Event;
+    use fe_semantics::namespace::types::{
+        Base,
+        FixedSize,
+    };
+    use yultsur::*;
+
+    #[test]
+    fn test_encode() {
+        assert_eq!(
+            encode(vec![FixedSize::Base(Base::U256)], vec![expression! { 42 }]).to_string(),
+            "abi_encode_uint256(42)"
```

Comment:
I guess this is another case where we will need to ensure generated method names can not clash with user defined method names. Maybe just generating them with a leading `$` would be sufficient for now? I'm going to create an issue for that.

Reply:
I created an issue for this https://github.com/ethereum/fe/issues/102